### ISSUE TITLE
python >3.8: ValueError: settimeout(): only 0 timeout is allowed on transport sockets

### ISSUE THREAD

**Issue by Slavix72** [komuw/naz#185]
python 3.8.0
ubuntu 18.04
naz 0.7.4
I create ussd client. After a client was started,  it had been terminated with error.

{"timestamp": "2019-11-27 16:06:16,041", "event": "naz.Client.dequeue_messages", "stage": "start", "smsc_host": "217.148.54.170", "system_id": "16500", "client_id": "QK5YN6ZNNAB9N4G0T", "pid": 64740}
{"timestamp": "2019-11-27 16:06:16,041", "event": "naz.Client.dequeue_messages", "stage": "start", "current_session_state": "CLOSED", "state": "awaiting naz to change session state to `BOUND_TRX`. sleeping for 30.00 seconds", "smsc_host": "217.148.54.170", "system_id": "16500", "client_id": "QK5YN6ZNNAB9N4G0T", "pid": 64740}
{"timestamp": "2019-11-27 16:06:16,042", "event": "naz.Client.receive_data", "stage": "start", "smsc_host": "217.148.54.170", "system_id": "16500", "client_id": "QK5YN6ZNNAB9N4G0T", "pid": 64740}
{"timestamp": "2019-11-27 16:06:16,042", "event": "naz.Client.receive_data", "stage": "end", "state": "naz is yet to bind to SMSC. sleeping for 30.00 seconds", "smsc_host": "217.148.54.170", "system_id": "16500", "client_id": "QK5YN6ZNNAB9N4G0T", "pid": 64740}
Traceback (most recent call last):
  File "/home/user/py/clear_ussd/uss.py", line 118, in <module>
    loop.run_until_complete(tasks)
  File "/home/user/Python-3.8.0/Lib/asyncio/base_events.py", line 608, in run_until_complete
    return future.result()
  File "/home/user/py/clear_ussd/venv/lib/python3.8/site-packages/naz/client.py", line 754, in connect
    sock.settimeout(self.socket_timeout)
  File "/home/user/Python-3.8.0/Lib/asyncio/trsock.py", line 188, in settimeout
    raise ValueError(
ValueError: settimeout(): only 0 timeout is allowed on transport sockets

After that I set
sock.settimeout(0) and client was started successfily.
Is it correct?

**Comment by komuw** [comment]
@Slavix72 
I can confirm that on python 3.8.0 this is the case. It is not the case however for python 3.7 which is what the tests run under.    

As a workaround, you can declare your `naz` client like; 
```python
import naz
client = naz.Client(
   ...
    socket_timeout=0.00,
)
```

**Comment by komuw** [comment]
Notes to self:
- https://github.com/komuw/naz/blob/c47f5030b720f3bac400dd6bd457b4415b0d5b7b/naz/client.py#L750-L757    

- `Changed in version 3.7: The method no longer toggles SOCK_NONBLOCK flag on socket.type.`; https://docs.python.org/3.8/library/socket.html#socket.socket.settimeout

**Comment by Slavix72** [comment]
i do not see a bug  with "socket_timeout=0.00"  (Python 3.8)

**Comment by komuw** [comment]
@Slavix72 
This is now fixed in version; v0.7.7
